### PR TITLE
Add checksum for py-argon2-cffi 21.1.0 and update python dependency

### PR DESCRIPTION
**Pull request by iarspider**

### CODE REVIEW

**Comment by iarspider** [comment]
@spackbot run pipeline

File: var/spack/repos/builtin/packages/py-argon2-cffi/package.py [hunk 1]
```diff
@@ -10,8 +10,11 @@ class PyArgon2Cffi(PythonPackage):
...
     pypi = "argon2-cffi/argon2-cffi-20.1.0.tar.gz"
 
+    version('21.1.0', sha256='f710b61103d1a1f692ca3ecbd1373e28aa5e545ac625ba067ff2feca1b2bb870')
     version('20.1.0', sha256='d8029b2d3e4b4cea770e9e5a0104dd8fa185c1724a0f01528ae4826a6d25f97d')
 
+    depends_on('python@2.7,3.5:', type=('build', 'run'))
+    depends_on('python@3.5:', type=('build', 'run'), when='@21.1.0:')
     depends_on('py-setuptools', type='build')
     depends_on('py-cffi@1.0.0:', type=('build', 'run'))
     depends_on('py-six', type=('build', 'run'))
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-argon2-cffi/package.py]: Six is no longer required in the latest version

File: var/spack/repos/builtin/packages/py-argon2-cffi/package.py [hunk 2]
```diff
@@ -10,8 +10,11 @@ class PyArgon2Cffi(PythonPackage):
     homepage = "https://argon2-cffi.readthedocs.io/"
     pypi = "argon2-cffi/argon2-cffi-20.1.0.tar.gz"
 
+    version('21.1.0', sha256='f710b61103d1a1f692ca3ecbd1373e28aa5e545ac625ba067ff2feca1b2bb870')
     version('20.1.0', sha256='d8029b2d3e4b4cea770e9e5a0104dd8fa185c1724a0f01528ae4826a6d25f97d')
 
+    depends_on('python@2.7,3.5:', type=('build', 'run'))
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-argon2-cffi/package.py]: @haampie does `2.7` work or does it need to be `2.7:2.8` or `2.7.0:2.7`?

**Review by scheibelp — DISMISSED** [review]

**Comment by scheibelp** [comment]
@adamjstewart apologies I missed that you self-assigned this: I will leave it for you to look at.

File: var/spack/repos/builtin/packages/py-argon2-cffi/package.py (hunk 2 above)
Comment (iarspider) [reply var/spack/repos/builtin/packages/py-argon2-cffi/package.py]: I think it does:
```
>>> argo = Spec('python@2.7,3.5:')
>>> pyth = Spec('python@2.7.8')
>>> pyth.satisfies(argo)
True
>>> pyth = Spec('python@2.6')
>>> pyth.satisfies(argo)
False
>>> pyth = Spec('python@3.3')
>>> pyth.satisfies(argo)
False
>>> pyth = Spec('python@3.5')
>>> pyth.satisfies(argo)
True
```

**Review by adamjstewart — APPROVED** [review]

**Pull request merged**